lib/elixir/lib/map_set.ex

```diff
@@ -32,8 +32,11 @@ defmodule MapSet do
   structures: for example, see `MapSet.new/1` or `Enum.into/2`.
   """
 
-  @opaque t :: %__MODULE__{map: map}
   @type value :: term
+
+  @opaque t(value) :: %__MODULE__{map: %{optional(value) => true}}
+  @opaque t :: t(term)
```

Comment:
`@type t` as discussed in https://github.com/elixir-lang/elixir/pull/5365#discussion_r85363628